ui/src/common/useSelectedFields.js

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react'
+import PropTypes from 'prop-types'
+import { useDispatch, useSelector } from 'react-redux'
+import { setToggleableFields } from '../actions'
+
+const useSelectedFields = ({ resource, columns }) => {
```

Comment:
It would be awesome to have unit tests for this :)

Reply:
I'll have to dig in a bit at testing :). I'll look at the examples and will write tests. Also I'm thinking of modifying the logic a bit more inorder to make it more compatible with future use cases. Will let you know soon! :crossed_fingers: